small-talk/components/smolAuth.js

```diff
@@ -0,0 +1,55 @@
+import { getCookie } from "@/util/smolCookie";
+
+export function smolAuth(getServerSidePropsFunc) {
+    return async (ctx) => {
+        const { req } = ctx;
+        
+        if (req.headers.cookie) {
+            const session_id = getCookie("session_id", req.headers.cookie);
+            //console.log(session_id)
+            try {
+                let response = await fetch("http://" + req.headers.host + "/api/auth2/", {
+                    method: "POST",
+                    headers: {
+                        "Content-Type" : "application/json",
+                        "Authorization" : session_id
+                    }
+                })
+
+                let json = (await response.json());
+                
+                if (json == null) {
```

Comment:
Would this ever be null if the json is either returned as found:false or found:true? Or would it return null if there was some sort of error in the mongodb connection?

Reply:
It would return null if there is an error in the response. false/true is if the session is valid or not.